third_party/nvfuser/test/test_gpu_combined_inner_outer_reduction.cpp

```diff
@@ -0,0 +1,568 @@
+  if (verbose)
+    std::cout << "inner_reduction_tv " << inner_reduction_tv->toString()
+              << std::endl;
+  auto reference_tv_inner =
+      reduction_scheduler_utils::sortAndRFactor(inner_reduction_tv);
+  if (verbose)
+    std::cout << "reference_tv_inner " << reference_tv_inner->toString()
+              << std::endl;
+
+  outer_reduction_tv->split(0, bidy, false);
+  auto partialResult = outer_reduction_tv->rFactor({1});
+  partialResult->cacheBefore();
+  partialResult->setMemoryType(MemoryType::Global);
+  auto partialResultReload = partialResult->cacheAfter();
+
+  outer_reduction_tv->split(0, nloady, false);
+  outer_reduction_tv->split(-1, tidx);
+  outer_reduction_tv->split(-2, bidy);
+  outer_reduction_tv->axis(1)->parallelize(ParallelType::TIDy);
+  outer_reduction_tv->axis(-2)->parallelize(ParallelType::BIDy);
+  outer_reduction_tv->axis(-1)->parallelize(ParallelType::TIDx);
+
+  if (verbose)
+    std::cout << "outer_reduction_tv " << outer_reduction_tv->toString()
+              << std::endl;
+  auto reference_tv_outer =
+      reduction_scheduler_utils::sortAndRFactor(outer_reduction_tv);
+  if (verbose)
+    std::cout << "reference_tv_outer " << reference_tv_outer->toString()
+              << std::endl;
+
+  reduction_scheduler_utils::propagateTransformation(
+      reference_tv_inner, {partialResultReload});
+  reduction_scheduler_utils::propagateTransformation(
+      reference_tv_outer, {partialResultReload});
+
+  std::vector<TensorView*> cached_gmem_temp{partialResult};
+  // cached_gmem is float, may use a different vectorization factor
+  for (auto tv : cached_gmem_temp) {
+    tv->split(-1, 4);
+    tv->axis(-1)->parallelize(ParallelType::Vectorize);
+  }
+
+  reduction_scheduler_utils::propagateParallelization(
+      &fusion,
+      inner_reduction_tv,
+      reference_tv_inner,
+      true,
+      true,
+      cached_inputs,
+      cached_outputs);
+  reduction_scheduler_utils::propagateParallelization(
+      &fusion,
+      outer_reduction_tv,
+      reference_tv_outer,
+      true,
+      true,
+      cached_inputs,
+      cached_outputs);
+
+  inlineMost();
+  LaunchParams launch_constraints;
+  constexpr int64_t maxrregcount = 64;
+  CompileParams compile_params{DataType::Int, maxrregcount, true};
+  if (verbose)
+    fusion.print();
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor tv_input = at::randn({dim0, dim1}, options);
+  auto tv_aten_output = tv_input.to(at::kFloat).sum({1});
+  at::Tensor tv_cg_output = at::empty({dim0}, options);
+
+  at::Tensor qv_cg_output = at::empty({dim1}, options);
+  auto qv_aten_output = tv_input.to(at::kFloat).sum({0});
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {tv_input}, launch_constraints, compile_params);
+  fe.runFusion(
+      {tv_input},
+      {tv_cg_output, qv_cg_output},
+      launch_constraints,
+      compile_params);
+  TORCH_CHECK(
```

Comment:
When debug the code, `allclose` is much easier to see which output is wrong. seems these two validating methods are used interchangeably in our code.
changed back to `testValidate`

Reply:
Please use `testValidate`. Some old tests may still use `allclose`, but I don't see any reason to use `allclose` directly. `testValidate` actually uses `allclose`. https://github.com/csarofeen/pytorch/blob/devel/third_party/nvfuser/test/test_gpu_validator.h#L394

The point of `testValidate` is it adjusts the tolerance threshold based on the actual expressions in the given fusion. Not perfect, but it's been working much more robustly than just using `allclose` with the default threshold. 

Since `allclose` is used internally, if there's anything that you only see with `allclose` that is useful debugging, it should be possible to make it available with `testValidate`. Please consider extending `testValidate`.